### PR TITLE
[UNR-4724] Fix SpatialTestRepNotify failing on native in 4.26

### PR DESCRIPTION
**Pull request by mironec**
#### Description
Fix failing test in 4.26, where the expected native behaviour changed.
Thankfully, the behaviour is now the same as ours.

#### Release note
N/A

#### Tests
This is fixing a test.

#### Documentation
N/A, but added comments explaining this.

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestRepNotify/SpatialTestRepNotify.cpp [hunk 1]
```diff
@@ -151,12 +151,20 @@ void ASpatialTestRepNotify::PrepareTest()
 			}
 
 			// We consciously differ from native UE here
-			if (GetNetDriver()->IsA(USpatialNetDriver::StaticClass()))
+			// Also, the native behaviour changed when going from 4.25 to 4.26.
+			// On older versions, we expect the old array to have 3 elements, but on Spatial and on native starting from 4.26, we expect 2 elements.
+#if ENGINE_MINOR_VERSION >= 26
+			bool bOldArrayShouldHaveTwoElements = true;
```
Comment (simonsarginson) [inline SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestRepNotify/SpatialTestRepNotify.cpp:157]: minor: I would prefer to use a const int that defines the number of array elements. Less fragile (no need to magic number the != below) and more expressive.

Comment (mironec) [reply SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestRepNotify/SpatialTestRepNotify.cpp:157]: Hmm, though I also do check the additional element when it is available...

**Review by simonsarginson — APPROVED** [review]
Minor comment but otherwise LGTM

File: SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestRepNotify/SpatialTestRepNotify.cpp (hunk 1 above)
Comment (mironec) [reply SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestRepNotify/SpatialTestRepNotify.cpp:157]: And this way also lets me include more info in the failure messages...

**Pull request merged**